tests/lava/magma/core/model/py/test_ports.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+import unittest
+import time
+from multiprocessing.managers import SharedMemoryManager
+import numpy as np
+
+from lava.magma.compiler.channels.interfaces import AbstractCspSendPort, \
+    AbstractCspRecvPort
+
+from lava.magma.core.model.py.ports import PyInPort, PyInPortVectorDense, \
+    PyOutPort, PyOutPortVectorDense
+
+from lava.magma.compiler.channels.pypychannel import PyPyChannel
+
+
+class MockInterface:
+    def __init__(self, smm):
+        self.smm = smm
+
+
+def get_channel(smm, data, size, name="test_channel") -> PyPyChannel:
+    mock = MockInterface(smm)
+    return PyPyChannel(
+        message_infrastructure=mock,
+        src_name=name,
+        dst_name=name,
+        shape=data.shape,
+        dtype=data.dtype,
+        size=size
+    )
+
+
+class TestPyPorts(unittest.TestCase):
+    def test_py_in_port_vector_dense_probe_single_csp_recv_port(self):
+        """Tests PyInPortVectorDense probe method for single recv_csp_port"""
+        smm = SharedMemoryManager()
+        try:
+            smm.start()
+
+            data = np.random.randint(5, size=(2, 4))
```

Comment:
Is it important that this is randomized? Random numbers in unit tests are problematic because the tests become non-deterministic and may sometimes pass or fail.
It seems, here, that you could just use a hard-coded values. Otherwise fix the random-seed.

Reply:
You're right, no need to use random numbers here.
This has been addressed in latest commit.